core/atomic_c11.c

```diff
@@ -48,7 +48,12 @@
 
 /* GCC documentation refers to the types as I1, I2, I4, I8, I16 */
 typedef uint8_t  I1;
+
+#if __SIZEOF_INT__ == 2
+typedef unsigned int I2;
+#else
 typedef uint16_t I2;
+#endif
```

Comment:
```suggestion
/* the builtins are declared with "unsigned int", but "uint16_t" is typedef'ed
 * to "short unsigned int" on most platforms where "sizeof(int) == 2." */
#if __SIZEOF_INT__ == 2
typedef unsigned int I2;
#else
typedef uint16_t I2;
#endif
```

Maybe add a comment, like is done for the special handling of "I4" below? From the outside, this changes seems to make no sense and an explaining comment can be helpful.